vi_editor.md

```diff
@@ -0,0 +1,35 @@
+# **Vi Text Editor**
+
+There are many ways to edit files in Unix. Editing files using the screen-oriented text editor **vi** is one of the best ways. This editor enables you to edit lines in context with other lines in the file.
+
+An improved version of the vi editor which is called the **VIM** has also been made available now. Here, VIM stands for **Vi IM** proved.
+
+vi is generally considered the de facto standard in Unix editors because −
+
+    * It's usually available on all the flavors of Unix system.
+
+    * Its implementations are very similar across the board.
+
+    * It requires very few resources.
+
+    * It is more user-friendly than other editors such as the **ed** or the **ex**.
+
+You can use the **vi** editor to edit an existing file or to create a new file from scratch. You can also use this editor to just read a text file.  
+
+# Operation modes
+While working with the vi editor, we usually come across the following two modes −
+
+  * **Command mode:**
+    This mode enables you to perform administrative tasks such as saving the files, executing the commands, moving the cursor, cutting (yanking) and pasting the lines or words, as well as finding and replacing. In this mode, whatever you type is interpreted as a command.
+
+  * **Insert mode:**  
+    This mode enables you to insert text into the file. Everything that's typed in this mode is interpreted as input and placed in the file.
+
+vi always starts in the **command mode.** To enter text, you must be in the insert mode for which simply type i. To come out of the insert mode, press the **Esc** key, which will take you back to the command mode.
+
+**Hint** − If you are not sure which mode you are in, press the Esc key twice; this will take you to the command mode. You open a file using the vi editor. Start by typing some characters and then come to the command mode to understand the difference.
+
+# Starting The Vi editor
+
+
```

Comment:
Complete the remaining document by adding interactive snap shots of vi editor working on local terminal.